.github/workflows/main.yml

```diff
@@ -235,7 +235,7 @@ jobs:
     uses: ./.github/workflows/build-macos.yml
     with:
       platform: macos-x64
-      xcode-toolset-version: '11.7'
+      xcode-toolset-version: '12.5.1'
```

Comment:
This and the other toolchain update proposed here are to avoid the
mentioned bug in Apple Clang, so that `__builtin_bit_cast` can be used
with that compiler.  Since I don't think we should be using the
builtin, I don't think this is needed either.  There may be other
reasons to move the GHA toolchain version forward, but that should be
a separate discussion.